src/main/java/com/bisnode/opa/client/OpaConfiguration.java

```diff
@@ -1,8 +1,47 @@
 package com.bisnode.opa.client;
 
-import lombok.Value;
+import java.beans.ConstructorProperties;
+import java.util.Objects;
 
-@Value
-public class OpaConfiguration {
-    String url;
+/**
+ * Contains all configuration needed to set up {@link OpaClient}
+ */
+public final class OpaConfiguration {
+    private final String url;
+
+    /**
+     * @param url base URL to OPA server, containing protocol, and port (eg. http://localhost:8181)
+     */
+    @ConstructorProperties({"url"})
+    public OpaConfiguration(String url) {
+        this.url = url;
+    }
+
+    /**
+     * @return url base URL to OPA server, containing protocol, and port
+     */
+    public String getUrl() {
+        return this.url;
+    }
+
+    public boolean equals(final Object o) {
+        if (o == this) return true;
+        if (!(o instanceof OpaConfiguration)) return false;
+        final OpaConfiguration other = (OpaConfiguration) o;
+        final Object this$url = this.getUrl();
```

Comment:
Dollar sign in Java? 🤔

Reply:
Yeah autogenerated lombok code is not the most beautiful one, i will change it.